Client/Frontend/Browser/MainMenuActionHelper.swift

```diff
@@ -65,6 +65,7 @@ class MainMenuActionHelper: PhotonActionSheetProtocol,
     weak var delegate: ToolBarActionMenuDelegate?
     weak var menuActionDelegate: MenuActionsDelegate?
     weak var sendToDeviceDelegate: SendToDeviceDelegate?
+    private var readerModeCache: ReaderModeCache = DiskReaderModeCache.sharedInstance
```

Comment:
I think it's best not to add this here. First, the cache needs to be a `MemoryReaderModeCache` whenever we're in private mode (we don't want to persist reader mode data to disk in private mode). Second, it also feels too much responsibility for the menu helper to hold a cache and determine what type of cache it should be. 

When debugging this, what have you found about `createRecordWithURL`? We currently discard it's return value, but in the failure use case explained in this ticket I guess it returns a "failure"? Can we confirm this, and then maybe investigate what is the failure?

Reply:
MainMenuActionHelper now uses `BrowserViewController`'s  `readerModeCache` so it fixes the responsibility of mem or disk cache.
By the way, when I debug `createRecordWithURL`, it returns success

Reply:
@lmarceau Any thoughts on new changes?

Reply:
Hi @ttuygun, haven't had the change to put time on this issue. I'll get back to you next week. Thanks!